cmd/controller/app/BUILD.bazel

```diff
@@ -32,10 +32,15 @@ go_library(
         "//pkg/util:go_default_library",
         "//pkg/util/feature:go_default_library",
         "@com_github_spf13_cobra//:go_default_library",
+        "@io_istio_client_go//pkg/apis/networking/v1alpha3:go_default_library",
+        "@io_istio_client_go//pkg/apis/networking/v1beta1:go_default_library",
+        "@io_istio_client_go//pkg/clientset/versioned:go_default_library",
+        "@io_istio_client_go//pkg/informers/externalversions:go_default_library",
```

Comment:
cc @meyskens did we ever come to a conclusion on depending on other packages like this? I'm nervous importing another client-go as it adds an additional dimension to our k8s dependencies (which are tricky enough as is!)

Reply:
Does Istio make just API types available in a module that does not depend on client-go anywhere? This would allow us to generate our own clientsets/informers (which is the recommended way to depend on external projects)

Reply:
Maybe this can be used: https://pkg.go.dev/istio.io/api?

Reply:
+ copying these files? (idk, this will probably overcome the client-go dependency; but seems not that elegant):
- https://github.com/istio/client-go/blob/master/pkg/clientset/versioned/scheme/register.gen.go
- https://github.com/istio/client-go/blob/master/pkg/informers/externalversions/networking/v1beta1/gateway.gen.go
- https://github.com/istio/client-go/blob/master/pkg/informers/externalversions/networking/v1beta1/virtualservice.gen.go
- https://github.com/istio/client-go/blob/master/pkg/informers/externalversions/networking/v1alpha3/gateway.gen.go
- https://github.com/istio/client-go/blob/master/pkg/informers/externalversions/networking/v1alpha3/virtualservice.gen.go

Reply:
I'll try to add the clientsets/informers generation